dia-batching-server/src/price_updater.rs

```diff
@@ -25,99 +25,55 @@ where
-	let Quotation { name, symbol, blockchain, price, time, volume_yesterday, .. } = value;
+	let Quotation { name, symbol, blockchain, price, time, supply, .. } = value;
 
 	let price = convert_decimal_to_u128(&price)?;
-	let supply = convert_decimal_to_u128(&volume_yesterday)?;
+	let supply = convert_decimal_to_u128(&supply)?;
 
 	let coin_info = CoinInfo {
 		name: name.into(),
 		symbol: symbol.into(),
 		blockchain: blockchain.unwrap_or("FIAT".to_string()).into(),
 		price,
-		last_update_timestamp: time.timestamp().unsigned_abs(),
+		last_update_timestamp: time,
 		supply,
 	};
 
 	info!("Coin Price: {:#?}", price);
-	info!("Coin Supply: {:#?}", volume_yesterday);
 	info!("Coin Info : {:#?}", coin_info);
 
 	Ok(coin_info)
 }
 
 async fn update_prices<T>(
 	coins: Arc<CoinInfoStorage>,
-	maybe_supported_currencies: &Option<HashSet<AssetSpecifier>>,
+	supported_currencies: &HashSet<AssetSpecifier>,
 	api: &T,
-	rate: std::time::Duration,
 ) where
-	T: DiaApi + Send + Sync + 'static,
+	T: PriceApi + Send + Sync + 'static,
 {
 	let mut currencies = vec![];
 
-	if let Ok(quotable_assets) = api.get_quotable_assets().await {
-		info!("No. of quotable assets to retrieve : {}", quotable_assets.len());
-
-		for quotable_asset in quotable_assets {
-			let asset = AssetSpecifier {
-				blockchain: quotable_asset.asset.blockchain.clone(),
-				symbol: quotable_asset.asset.symbol.clone(),
-			};
-
-			if maybe_supported_currencies
-				.as_ref()
-				.map_or(true, |supported| supported.contains(&asset))
-			{
-				match api.get_quotation(&quotable_asset).await.and_then(convert_to_coin_info) {
-					Ok(coin_info) => {
-						currencies.push(coin_info);
-					},
-					Err(err) => {
-						error!("Error while retrieving quotation for {:?}: {}", quotable_asset, err)
-					},
-				}
-				tokio::time::delay_for(rate).await;
-			}
-		}
-	}
+	let supported_currencies = supported_currencies.iter().collect::<Vec<_>>();
 
-	if let Some(supported_currencies) = maybe_supported_currencies.as_ref() {
-		for asset in supported_currencies.iter() {
-			// We do support both these 'blockchain' identifiers while DIA doesn't provide data for them
-			if asset.blockchain == "FIAT" || asset.blockchain == "Amplitude" {
-				// Create dummy QuotedAsset. We only need it to have the symbol and blockchain
-				let quoted_asset = QuotedAsset {
-					asset: Asset {
-						symbol: asset.symbol.clone(),
-						name: "".to_string(),
-						address: "".to_string(),
-						decimals: 0,
-						blockchain: asset.blockchain.clone(),
-					},
-					volume: Default::default(),
-				};
-				match api.get_quotation(&quoted_asset).await.and_then(convert_to_coin_info) {
-					Ok(coin_info) => {
-						currencies.push(coin_info);
-					},
-					Err(err) => {
-						error!("Error while retrieving quotation for {:?}: {}", quoted_asset, err)
-					},
-				}
-			}
-		}
-	}
+	api.get_quotations(supported_currencies)
+		.await
+		.unwrap_or_default()
```

Comment:
While I understand the reason behind never returning an error, we could also have `fn get_quotation()` to simply return `Vec<Quotation>`.

We can still log the errors with the method returning an empty vec.

Reply:
I changed the signature of the function to reflect that it doesn't return an Error